getting-started/src/main/kotlin/com/google/cloud/kotlin/demo/EmojifyController.kt

```diff
@@ -0,0 +1,160 @@
+import com.google.cloud.vision.v1.*
+import com.google.cloud.vision.v1.Feature.Type
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.awt.Polygon
+import java.io.FileOutputStream
+import java.io.PrintStream
+import java.nio.file.Files
+import java.nio.file.Paths
+import javax.imageio.ImageIO
+
+val storage: Storage = StorageOptions.getDefaultInstance().service
+val bucket = storage.get("cloud-kotlin-samples")
+val bucketUri = "gs://cloud-kotlin-samples"
+
+const val RESOURCES_PATH = "resources/"
+const val EMOJIS_PATH = RESOURCES_PATH + "emojis/"
+
+enum class Emoji {
+    JOY, ANGER, SURPRISE, SORROW, NONE
+}
+
+val emojiPic = mapOf(
+    Emoji.JOY to "joy.png",
+    Emoji.ANGER to "anger.png",
+    Emoji.SURPRISE to "surprise.png",
+    Emoji.SORROW to "sorrow.png",
+    Emoji.NONE to "none.png"
+)
+
+//Returns best emoji based on detected emotions likelihoods
+fun bestEmoji(annotation: FaceAnnotation): Emoji {
+    val emotionsLikelihood: Array<Likelihood> = arrayOf(Likelihood.VERY_LIKELY, Likelihood.LIKELY, Likelihood.POSSIBLE)
+    val emotions = mapOf(
+        Emoji.JOY to annotation.joyLikelihood,
+        Emoji.ANGER to annotation.angerLikelihood,
+        Emoji.SURPRISE to annotation.surpriseLikelihood,
+        Emoji.SORROW to annotation.sorrowLikelihood
+    )
+
+    for (emotionLikelihood in emotionsLikelihood) {
+        for (emotion in emotions) {
+            if (emotion.value == emotionLikelihood) return emotion.key
+        }
+    }
+    return Emoji.NONE
+}
+
+//Downloads source image to a temp file
+fun downloadObject(objectName: String) {
+    val blob = bucket.get(objectName)
+    val localFilePath = Paths.get("/tmp/$objectName")
+    val writeTo = PrintStream(FileOutputStream(localFilePath.toFile()))
+    writeTo.write(blob.getContent())
+    writeTo.close()
+}
+
+@RestController
+class EmojifyController {
+    @GetMapping("/emojify")
+    fun emojify(@RequestParam(value = "objectName") objectName: String): String {
+
+        var publicUrl:String =
+            "https://storage.googleapis.com/${bucket.name}/emojified/emojified-$objectName" // api response
+
+        val requests = ArrayList<AnnotateImageRequest>()
+        //Setting up image annotation request
+        val vision = ImageAnnotatorClient.create()
+        val source = ImageSource.newBuilder().setGcsImageUri("$bucketUri/$objectName").build() //Fetching our source image
+        val img = Image.newBuilder().setSource(source).build()
+        val feat = Feature.newBuilder().setType(Type.FACE_DETECTION).build()
+        val request = AnnotateImageRequest.newBuilder()
+                .addFeatures(feat)
+                .setImage(img)
+                .build()
+
+        requests.add(request)
+
+        // Calls vision api on above image annotation requests
+        val response = vision.batchAnnotateImages(requests)
```

Comment:
Do we need to use batch? Looks like you are only doing one image at a time

Reply:
Answer: Yes, we need to use batch (no single image method provided in the Java client)

Reply:
Answer: Yes we do, the java client doesn't have a single image method for some reason